File: android-client/app/src/main/java/com/nexus/platform/api/StorageApi.kt
package com.nexus.platform.api

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson
import com.google.gson.JsonObject

/**
 * 存储API处理器，提供本地数据存储功能
 */
class StorageApi(private val context: Context) : ApiHandler {
    private val prefs: SharedPreferences = context.getSharedPreferences("wx_storage", Context.MODE_PRIVATE)
    private val gson = Gson()

    override suspend fun handle(api: String, params: JsonObject): Any? {
        return when (api) {
            "wx.setStorage", "wx.setStorageSync" -> setStorage(params)
            "wx.getStorage", "wx.getStorageSync" -> getStorage(params)
            "wx.removeStorage", "wx.removeStorageSync" -> removeStorage(params)
            "wx.clearStorage", "wx.clearStorageSync" -> clearStorage()
            else -> null
        }
    }

    /**
     * 设置存储数据
     * @param params 包含key和data的参数
     * @return 操作结果
     */
    private fun setStorage(params: JsonObject): Map<String, String> {
        val key = params.get("key")?.asString ?: return mapOf("errMsg" to "setStorage:fail")
        val data = params.get("data")
        
        prefs.edit().putString(key, gson.toJson(data)).apply()
        return mapOf("errMsg" to "setStorage:ok")
    }

    /**
     * 获取存储数据
     * @param params 包含key的参数
     * @return 存储的数据
     */
    private fun getStorage(params: JsonObject): Map<String, Any?> {
        val key = params.get("key")?.asString ?: return mapOf("errMsg" to "getStorage:fail")
        val data = prefs.getString(key, null)
        
        return if (data != null) {
            mapOf(
                "data" to gson.fromJson(data, Any::class.java),
                "errMsg" to "getStorage:ok"
            )
        } else {
            mapOf("errMsg" to "getStorage:fail")
        }
    }

    /**
     * 删除存储数据
     * @param params 包含key的参数
     * @return 操作结果
     */
    private fun removeStorage(params: JsonObject): Map<String, String> {
        val key = params.get("key")?.asString ?: return mapOf("errMsg" to "removeStorage:fail")
        prefs.edit().remove(key).apply()
        return mapOf("errMsg" to "removeStorage:ok")
    }

    /**
     * 清空所有存储数据
     * @return 操作结果
     */
    private fun clearStorage(): Map<String, String> {
        prefs.edit().clear().apply()
        return mapOf("errMsg" to "clearStorage:ok")
    }
}

File: ios-client/NexusPlatform/Sources/NexusPlatform/API/StorageApi.swift
import Foundation

class StorageApi: ApiHandler {
    private let userDefaults = UserDefaults.standard
    private let keyPrefix = "nexus.bridge.storage."
    
    func handle(api: String, params: [String: Any]) async throws -> Any {
        switch api {
        case "wx.setStorageSync", "wx.setStorage":
            return try setStorage(params)
        case "wx.getStorageSync", "wx.getStorage":
            return try getStorage(params)
        case "wx.removeStorageSync", "wx.removeStorage":
            return try removeStorage(params)
        case "wx.clearStorageSync", "wx.clearStorage":
            return clearStorage()
        default:
            throw ApiError.notImplemented
        }
    }
    
    private func setStorage(_ params: [String: Any]) throws -> [String: String] {
        guard let key = params["key"] as? String else {
            throw ApiError.invalidParameter
        }
        
        if let data = params["data"] {
            let jsonData = try JSONSerialization.data(withJSONObject: data)
            userDefaults.set(jsonData, forKey: namespacedKey(for: key))
        }
        
        return ["errMsg": "setStorage:ok"]
    }
    
    private func getStorage(_ params: [String: Any]) throws -> [String: Any] {
        guard let key = params["key"] as? String else {
            throw ApiError.invalidParameter
        }
        
        guard let jsonData = userDefaults.data(forKey: namespacedKey(for: key)),
              let data = try? JSONSerialization.jsonObject(with: jsonData) else {
            return ["errMsg": "getStorage:fail"]
        }
        
        return [
            "data": data,
            "errMsg": "getStorage:ok"
        ]
    }
    
    private func removeStorage(_ params: [String: Any]) throws -> [String: String] {
        guard let key = params["key"] as? String else {
            throw ApiError.invalidParameter
        }
        
        userDefaults.removeObject(forKey: namespacedKey(for: key))
        return ["errMsg": "removeStorage:ok"]
    }
    
    private func clearStorage() -> [String: String] {
        let dictionary = userDefaults.dictionaryRepresentation()
        dictionary.keys.filter { $0.hasPrefix(keyPrefix) }.forEach { key in
            userDefaults.removeObject(forKey: key)
        }
        return ["errMsg": "clearStorage:ok"]
    }

    private func namespacedKey(for rawKey: String) -> String {
        keyPrefix + rawKey
    }
}

enum ApiError: Error {
    case notImplemented
    case invalidParameter
}
